addons/project/models/project_project_stage.py

```diff
@@ -41,6 +42,19 @@ def unlink_wizard(self, stage_view=False):
         }
 
     def write(self, vals):
+        if vals.get('company_id'):
+            company = self.env['res.company'].browse(vals.get('company_id'))
+            # Checking if there is a project with a different company_id than the target one. If so raise an error since this is not allowed
+            project = self.env['project.project'].search(['&', ('stage_id', 'in', self.ids), ('company_id', '!=', vals.get('company_id'))], limit=1)
+
+            if project:
+                raise exceptions.UserError(
+                    _("You are not able to switch the company of this stage to %s since it currently "
+                    "includes projects associated with %s. Please ensure that this stage exclusively "
+                    "consists of projects linked to %s.",
+                    company.name, project.company_id.name if project.company_id else "no company", company.name)
```

Comment:
```suggestion
                    _("You are not able to switch the company of this stage to %(company_name)s since it currently "
                    "includes projects associated with %(project_company_name)s. Please ensure that this stage exclusively "
                    "consists of projects linked to %(company_name)s.",
                        company_name=company.name,
                        project_company_name=project.company_id.name or "no company"
                    )
```
Isn't that possible?
Also, `project.company_id.name or "no company"` should be enough. :thinking:

Reply:
Yes seems possible :+1: thanks!